iFixit/checkout-ui-extensions
# Shipping fallback notes

Closes https://github.com/iFixit/checkout-ui-extensions/issues/31

Implements the feature from shopify-v2 as described in https://github.com/iFixit/shopify-v2/issues/526

The translations should match [those from shopify-v2](https://github.com/iFixit/shopify-v2/blob/4e1c1341fb89d66c697f934b6b9b127e0be1ae47/src/locales/en.default.json#L606-L614).

## QA

Run local dev and preview `purchase.checkout.shipping-option-list.render-before`. If your cart meets any of the conditions outlined in https://github.com/iFixit/shopify-v2/issues/526, then you should see a warning banner with extra information like below. Please test all the conditions.

![image](https://github.com/iFixit/checkout-ui-extensions/assets/52104630/a5cf08bb-08da-4f11-b403-f0cf4e179c96)

You might have to play with the [shipping rates](https://admin.shopify.com/store/ifixit-checkout-dev/settings/shipping) on the checkout dev store in order make it such that no shipping options show up. Alternatively, you could drop that check from the code with a diff like:

```diff
-   if (!isHeadlessStore(shop) || hasDeliveryOptions) {
+   if (!isHeadlessStore(shop)) {
      return null;
   }
```

You might also need to make certain products in stock, change their weights, or change their tags. That is all fine, it's a dev store.
